### PR TITLE
First version of native packaging scripts that produces .deb packages fo...

### PR DESCRIPTION
**Pull request by BrainFlame**
...r Debian.

### CODE REVIEW

File: scripts/conf/treode.init [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+# Short-Description: Treode server
+# Description:       Treode server
+### END INIT INFO
+
+# Author: treode@treode.com
+
+# PATH should only include /usr/* if it runs after the mountnfs.sh script
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+
+#JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:/bin/java::")
```
Comment (Topher-the-Geek) [inline scripts/conf/treode.init]: This commented line isn't helpful. Can we remove it? Or, explain why we might uncomment it in the future.

File: scripts/conf/treode.init [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+# Define LSB log_* functions.
+# Depend on lsb-base (>= 3.0-6) to ensure that this file is present.
+. /lib/lsb/init-functions
+
+#
+# Function that starts the daemon/service
+#
+do_start()
+{
+# Create 3KV_DB if does not exist
```
Comment (Topher-the-Geek) [inline scripts/conf/treode.init]: Can we indent this like the rest of the function? I know it's shell, so sometimes small things like that...

File: scripts/conf/treode.init [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	#   2 if daemon could not be started
+	start-stop-daemon --start --quiet --pidfile $PIDFILE --exec $DAEMON --test > /dev/null \
+		|| return 1
+	start-stop-daemon --start --pidfile $PIDFILE --make-pidfile \
+	    --exec $DAEMON \
+	    --background -- \
+	    $DAEMON_ARGS \
+		|| return 2
+
+#	    --startas /bin/sh -- -c "exec $DAEMON $DAEMON_ARGS > /opt/treode/logs/treode.log 2>&1" \
```
Comment (Topher-the-Geek) [inline scripts/conf/treode.init]: Similarly to the commented JAVA_HOME, this isn't helpful. Let's remove it, or explain what it's for.

File: scripts/package.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+echo "PACKAGING TREODE FOR Debian..."
+
+STAGEDIR0=/var/tmp/treode.$$
+STAGEDIR=${STAGEDIR0}/treode-0.2.0
+mkdir -p ${STAGEDIR}
+
+cp ${SCRIPT_DIR}/conf/treode.init ${STAGEDIR}
+cd ${STAGEDIR}
+
+wget https://oss.treode.com/examples/finagle/0.2.0/finagle-server-0.2.0.jar
```
Comment (Topher-the-Geek) [inline scripts/package.sh]: Shouldn't we get the .jar from the build? After build-merge script runs, it's available under examples/finagle/target

File: scripts/conf/treode.init [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+# Provides:          treode
+# Required-Start:    $network $local_fs
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: Treode server
+# Description:       Treode server
+### END INIT INFO
+
+# Author: treode@treode.com
```
Comment (Topher-the-Geek) [inline scripts/conf/treode.init]: We don't have an alias setup for treode@treode.com. We do have questions@treode.com

File: scripts/conf/treode.init [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+# PATH should only include /usr/* if it runs after the mountnfs.sh script
+PATH=/sbin:/usr/sbin:/bin:/usr/bin
+
+#JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:/bin/java::")
+JAVA_HOME=/usr/lib/jvm/default-java
+DESC=treode            
+NAME=treode            
+DAEMON="$JAVA_HOME/bin/java"
+TREODEKV_DB=/opt/treode/db/store.3kv
+DAEMON_ARGS="-jar /opt/treode/lib/finagle-server-0.2.0.jar serve -solo  -log.output=/opt/treode/db/treode.log $TREODEKV_DB"
```
Comment (Topher-the-Geek) [inline scripts/conf/treode.init]: The version number 0.2.0 appears throughout frequently. We need an easy way to upgrade versions.

**Comment by Topher-the-Geek** [comment]
It's exciting to see this come together.

It is clear that we should meet do discuss how we want to hook all this into the build/release process.

**Comment by Topher-the-Geek** [comment]
This latest CL is a great step improvement.

I see the the option in package.sh to use the jar from the build over wget. I'm still wondering, why allow wget at all? I think that ultimately we want to publish only the .deb and not publish .jar at all. So ultimately, there will be no jar on oss.treode.com to wget.

I see the 0.2.0 changed to PKGVER in package.sh and treode.rules. What can we do about 0.2.0 in treode.init?

**Comment by BrainFlame** [comment]
I think that allowing to package .deb and test different versions published in oss.treode.com via ivy is useful for developers.

**Pull request closed without merging**